resourceLock.lock();
try {
  doSomething();
}
catch (Exception e) {
  logError("An error occurred calling doFinally()");
}
finally {
  resourceLock.unlock();
}
